src/position.cpp

```diff
@@ -825,6 +825,41 @@ string Position::fen(bool sfen, bool showPromoted, int countStarted, std::string
 }
 
 
+/// Position::fog_fen() returns a Fog of War (dark chess) FEN representation
+/// of the position. Squares where current player can't move to are filled with
+/// NO_PIECE (wall squares)
+
+string Position::fog_fen(bool sfen, bool showPromoted, int countStarted, std::string holdings) {
+  Color us = sideToMove;
+  Bitboard fog;
+  
+  // Our own pieces are visible
+  Bitboard visible = pieces(us);
+
+  // Squares where we can move to are visible as well
+  for (const auto& m : MoveList<LEGAL>(*this))
+  {
+    Square to = to_sq(m);
+    visible |= to;
+  }
+
+  // Everything else is invisible
+  fog = ~visible & board_bb();
+
+  // Fill in invisible squares with walls
+  while (fog)
+  {
+    Square sq = pop_lsb(fog);
+    if (piece_on(sq)) remove_piece(sq);
+
+    st->wallSquares |= sq;
```

Comment:
This method looks getter-like from its name, but seems to mess with the state here. Would be nicer to make this method `const` just like `fen()` itself.

Reply:
First I made it const, but because remove_piece itself is not const, compilation failed.

Reply:
Isn't that a problem that I have to replicate almost all of the code of `fen()`?

Reply:
Implemented your bitboard argument suggestion.